New version: Microsoft.DSC version 3.1.0-preview.1

Gijsreyn opened the pull request:
Checklist for Pull Requests
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/241465)

Gijsreyn committed:
New version: Microsoft.DSC version 3.1.0-preview.1

Gijsreyn commented:
@mdanish-kh, out of curiosity. In #237024, you only specified one URL. Shouldn't the aarch64 have been included also?

mdanish-kh commented:
@Gijsreyn - I missed adding that it seems :D

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.installer.yaml:
```diff
@@ -0,0 +1,23 @@
+# Created using wingetcreate 1.9.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.DSC
```
```suggestion
PackageIdentifier: Microsoft.DSC.Preview
```

This is a preview version that should be added under `Microsoft.DSC.Preview` Package ID

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+# Created using wingetcreate 1.9.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.DSC
```
```suggestion
PackageIdentifier: Microsoft.DSC.Preview
```

This is a preview version that should be added under `Microsoft.DSC.Preview` Package ID

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.yaml:
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.9.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.DSC
```
```suggestion
PackageIdentifier: Microsoft.DSC.Preview
```

This is a preview version that should be added under `Microsoft.DSC.Preview` Package ID

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.installer.yaml:
```diff
@@ -0,0 +1,23 @@
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: dsc.exe
+  PortableCommandAlias: dsc
```
```suggestion
  PortableCommandAlias: dsc-preview
```

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.installer.yaml:
```diff
@@ -0,0 +1,23 @@
+- RelativeFilePath: dsc.exe
+  PortableCommandAlias: dsc
+Commands:
+- dsc
```
```suggestion
- dsc-preview
```

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+Publisher: Microsoft Corporation
+PublisherUrl: https://github.com/PowerShell
+PublisherSupportUrl: https://github.com/PowerShell/DSC/issues
+PackageName: DSC v3
```
```suggestion
PackageName: DSC v3 Preview
```

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+  - Author resources to manage your systems in any language
+  - Invoke individual resources
+  - Create configuration documents that define the desired state of a system
+Moniker: dscv3
```
```suggestion
Moniker: dscv3-preview
```

mdanish-kh commented on manifests/m/Microsoft/DSC/3.1.0-preview.1/Microsoft.DSC.installer.yaml:
```diff
@@ -0,0 +1,23 @@
+- dsc.yaml
+- winget
+Installers:
+- Architecture: x64
```
We should also include the arm64 variant (i should open a PR to add that in the stable manifest too)

Gijsreyn replied:
@mdanish-kh I have included the ARM64 variant also and changed the version number. The directory has changed, and yes, I also did it already for the stable manifest. Can you take a look again for me :)? Thanks a lot.

mdanish-kh reviewed (changes requested):
There are a few challenges with offering multiple channels (preview, stable) of a portable offering that use the same symlink. As per community's discussion on  [Element Chat](https://matrix.to/#/!LEwByvxhTNZZcKeCCY:gitter.im/$0aQdoR8IT-X1-MhU2TfDgN4HWizaLrvMcAfStnjCPSo?via=gitter.im&via=matrix.org&via=mozilla.org),  the consensus seems to be that we should use different metadata that includes a separate symlink, PackageName, Moniker etc. It still wouldn't be _ideal_ as noted in the discussion thread, but I guess would be the best UX we can offer unless channels support is available in the CLI

- https://github.com/microsoft/winget-cli/issues/147

---

Since I've suggested to alter the PackageID -

After committing the suggested changes, you would also have to modify the directory structure to match the preview PackageIdentifier. The manifest would be moved under the `manifests/m/Microsoft/DSC/Preview` directory. Subsequently, the yaml file names would also be changed to match the new Package ID.

Gijsreyn committed:
Merge branch 'microsoft:master' into Microsoft.DSC-3.1.0-preview.1-34441b0a-830d-4d1b-bf4d-f484ad0e534a

Gijsreyn committed:
Resolve remarks

mdanish-kh commented on manifests/m/Microsoft/DSC/3.0.0/Microsoft.DSC.installer.yaml:
```diff
@@ -18,6 +18,9 @@ Installers:
 - Architecture: x64
   InstallerUrl: https://github.com/PowerShell/DSC/releases/download/v3.0.0/DSC-3.0.0-x86_64-pc-windows-msvc.zip
   InstallerSha256: 16989F94FEB276B1F24BD3B6D69483DE631CC8A38110B0EC559F56B2E09E9F17
+- Architecture: arm64
+  InstallerUrl: https://github.com/PowerShell/DSC/releases/download/v3.0.0/DSC-3.0.0-aarch64-pc-windows-msvc.zip
+  InstallerSha256: 29DBCC7DB22EFB43D54E3907ABA87816D9B8881BC03BA4CA75AE1162003318C5
```
Validation pipelines can only process a change in a single manifest per PR. This would need to be done in a separate PR

Gijsreyn replied:
I feel sad. I'll remove it in this PR.

mdanish-kh reviewed (changes requested):
> Compared to the other version, this is not the commandAlias.

@Gijsreyn That version would probably be removed from the repository (or atleast the portable variants) because of the issue you spotted with it not working with symlinks https://github.com/PowerShell/DSC/issues/618. Once the winget-cli pipelines have moved over to using the stable variant, for which I believe a PR is open right now, we can think of making that change. I didn't modify the last version until the pipelines have completely moved over. We should also probably hold merging this PR in before those changes have been merged over at winget-cli.

Regardless, the portable alias and the metadata should still be changed in this PR

mdanish-kh commented on manifests/m/Microsoft/DSC/Preview/3.1.0.0/Microsoft.DSC.installer.yaml:
```diff
@@ -0,0 +1,26 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.DSC.Preview
+PackageVersion: 3.1.0.0
```
```suggestion
PackageVersion: 3.1.0-preview.1
```

mdanish-kh commented on manifests/m/Microsoft/DSC/Preview/3.1.0.0/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.DSC.Preview
+PackageVersion: 3.1.0.0
```
```suggestion
PackageVersion: 3.1.0-preview.1
```

mdanish-kh commented on manifests/m/Microsoft/DSC/Preview/3.1.0.0/Microsoft.DSC.yaml:
```diff
@@ -0,0 +1,8 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.DSC.Preview
+PackageVersion: 3.1.0.0
```
```suggestion
PackageVersion: 3.1.0-preview.1
```

mdanish-kh reviewed (commented):
The version you used before was in fact the correct version. I know you probably tried to be consistent with the older version, but that's because that version included an MSIX and pipelines restrict the `PackageVersion` value to match the value coming from MSIX

Not sure when these versions will get an MSIX offering, but if they do, we can always open another PR to add the MSIX, change PackageVersion field that matches the version from MSIX and add `DisplayVersion` fields for portable installer types. That way, the correlation will still work for users on the portable installer type

Gijsreyn committed:
More resolve remarks

Gijsreyn commented:
> The version you used before was in fact the correct version. I know you probably tried to be consistent with the older version, but that's because that version included an MSIX and pipelines restrict the `PackageVersion` value to match the value coming from MSIX
> 
> Not sure when these versions will get an MSIX offering, but if they do, we can always open another PR to add the MSIX, change PackageVersion field that matches the version from MSIX and add `DisplayVersion` fields for portable installer types. That way, the correlation will still work for users on the portable installer type

I just changed the version back. I wasn't sure if the folder had to be changed also. If so, just let me know. The aliases are also changed.

mdanish-kh commented on manifests/m/Microsoft/DSC/Preview/3.1.0.0/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+Publisher: Microsoft Corporation
+PublisherUrl: https://github.com/PowerShell
+PublisherSupportUrl: https://github.com/PowerShell/DSC/issues
+PackageName: DSC v3
```
```suggestion
PackageName: DSC v3 Preview
```

This is needed so that the CLI can distinguish between the installations of DSC stable and DSC preview

mdanish-kh commented on manifests/m/Microsoft/DSC/Preview/3.1.0.0/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+  - Author resources to manage your systems in any language
+  - Invoke individual resources
+  - Create configuration documents that define the desired state of a system
+Moniker: dscv3
```
```suggestion
Moniker: dscv3-preview
```

so that users can just do `winget install dscv3-preview` to get the preview version

mdanish-kh commented:
> I wasn't sure if the folder had to be changed also (probably it has to become 3.1.0)

The version folder needs to be renamed, not to `3.1.0`, but the exact value of PackageVersion i.e,. `3.1.0-preview.1`

Gijsreyn committed:
Rename folder

mdanish-kh commented on manifests/m/Microsoft/DSC/Preview/3.1.0-preview.1/Microsoft.DSC.locale.en-US.yaml:
```diff
@@ -0,0 +1,41 @@
+  - Invoke individual resources
+  - Create configuration documents that define the desired state of a system
+Moniker: dscv3-preview
+Tags:
+- dsc
+- dscv3-preview
```
```suggestion
Tags:
- dsc
- dscv3
- dscv3-preview
```

Looks good, just one minor comment. I would still want this package to appear based on the tags field when someone does `winget search dscv3`

![image](https://github.com/user-attachments/assets/4dc8065b-6092-48a6-9040-3e588112dea2)

mdanish-kh commented:
@wingetbot run

Gijsreyn committed:
Change tag

mdanish-kh commented:
@wingetbot run

mdanish-kh reviewed (commented):
The yaml files need to be renamed to match the PackageID

`Microsoft.DSC.installer.yaml` -> `Microsoft.DSC.Preview.installer.yaml`
`Microsoft.DSC.locale.en-US.yaml` -> `Microsoft.DSC.Preview.locale.en-US.yaml`
`Microsoft.DSC.yaml` -> `Microsoft.DSC.Preview.yaml`

Gijsreyn committed:
Rename files

Gijsreyn commented:
@mdanish-kh I was looking at the message from the bot, but I don't really get it. What is missing here?

mdanish-kh commented:
@wingetbot run

Gijsreyn commented:
I guess it looks good now!
